src/sentry/tasks/sentry_apps.py

```diff
@@ -386,6 +381,9 @@ def send_and_save_webhook_request(sentry_app, app_platform_event, url=None):
         elif resp.status_code == 504:
             raise ApiTimeoutError.from_request(resp.request)
 
+        if 400 <= resp.status_code <= 404:
+            raise ClientError("Client Error", response=resp)
```

Comment:
@MeredithAnya Do we need to pass in a string like this to the constructor?

Reply:
@scefali we may not need to but I was trying to kind of mimic what requests does for `raise_for_status` https://github.com/psf/requests/blob/4f6c0187150af09d085c03096504934eb91c7a9e/requests/models.py#L936-L943

Reply:
@MeredithAnya @MeredithAnya Got it. It looks like we will lose some information about what the client error compared to `raise_for_status`. But at the very least I think we should include the staus code and URL. I'm thinking something like this:
```
class ClientError(RequestException):
    def __init__(self, status_code, url, response=None):
       http_error_msg = u'%s Client Error: for url: %s' % (status_code, url)
       return super(ClientError, self).__init__(http_error_msg, response=response)
```
This way the constructor handles constructing the string

Reply:
hokay yeah that wfm